bump pyOpenSSL req 0.13 to 0.14 to work with modern (~ 1.0.2) openssl versions

jantman opened the pull request:
Helga won't install anymore on my Arch Linux (rolling release) system, because pyOpenSSL <= 0.13 is incompatible with newer OpenSSL versions (I don't have the exact version that introduced the problem, but I'm running 1.0.2.a-1).

see https://github.com/pyca/pyopenssl/issues/276

FWIW, I couldn't find any documentation for Twisted on exactly what pyOpenSSL version it requires :(

shaunduncan commented:
Seems reasonable to me, but I did run into an issue with the vagrant setup I have for testing. It looks like 0.14 of pyOpenSSL has some new dependencies, notably cffi which requires libffi6 and libffi-dev. Would you mind adding a comment in requirements.txt and a mention in docs/source/getting_started.rst (much like it is for openssl and libssl-dev)? Also, the Vagrantfile might also need to go ahead and install those with openssl and libssl-dev as well (just so vagrant up continues to work).

jantman commented:
Yeah, sorry, I guess this was a bit slipshod. It's probably not going to be until this afternoon, but I'll do all of the above and confirm that the Vagrantfile works properly.

jantman commented:
Ok, this should be fixed with the latest commit pushed up. 

* I had to install git in the vagrant box as well, as ``pip freeze`` seems to fail if it can't find `git`.
* While doing that I realized that the venv was running with an amazingly archaic ``pip 1.1``, so I added a line to the Vagrantfile to upgrade pip in the venv.
* Fixed a line in the vagrantfile where "activate" was misspelled, that caused the venv to not activate on login.

shaunduncan commented:
Nice! I'll see about getting this packaged up for a patch release today. Thanks!

shaunduncan commented:
Just released helga 1.7.2 to include this change
